client/components/marketing-survey/cancel-purchase-form/index.jsx

```diff
@@ -267,8 +267,13 @@ class CancelPurchaseForm extends React.Component {
 	};
 
 	downgradeClick = () => {
-		this.props.downgradeClick();
-		this.recordEvent( 'calypso_purchases_downgrade_form_submit' );
+		if ( ! this.state.isSubmitting ) {
+			this.props.downgradeClick();
+			this.recordEvent( 'calypso_purchases_downgrade_form_submit' );
+			this.setState( {
```

Comment:
I haven't used this before, but I would think `isSubmitting` should be set to `true` immediately _before_ this function does the work it's supposed to do. Instead of after.

Also, does setting `isSubmitting` back to `false` get handled automatically? It seems like it from the few examples I've found

Reply:
Testing worked great however. I tried to quickly click twice but it didn't let me.

Reply:
Thanks, in case the cancelation fails, the dialog should close, so I'm not sure anything else needs to be done here. I think it's fine.